src/test/java/HandleGetRequestTest.java

```diff
@@ -2,9 +2,9 @@
 import java.util.*;
 import java.io.IOException;
 
-public class HttpHandlerSelectorTest extends junit.framework.TestCase {
+public class HandleGetRequestTest extends junit.framework.TestCase {
 
-  HttpHandlerSelector handler;
+  HandleGetRequest getRequest;
```

Comment:
So the `HandleGetRequest` class is kind of like a Response Factory that builds responses from GET requests. Would a name like `ResponseFactoryGetRequest` make more sense?

If so I would also rename the base class `HttpHandleRequest` to `ResponseFactory`